### PR TITLE
MEJO-23 Make bluetooth module work

### PR DESCRIPTION
**Pull request by bengyel**

### CODE REVIEW

File: Nucleo_F746ZG_app/Src/main.c [hunk 1]
```diff
@@ -293,7 +295,7 @@ void HAL_UART_RxCpltCallback(UART_HandleTypeDef *huart)
   {
     rplidar_rx_interrupt_handler();
   }
-  if(huart->Instance == USART3){
+  if(huart->Instance == USART3 || USART6){
```
Comment (johnyka) [inline Nucleo_F746ZG_app/Src/main.c]: This condition will be always true this way. Pls fix it

File: Nucleo_F746ZG_app/Src/utility.c [hunk 2]
```diff
@@ -27,12 +28,16 @@ void StartPrinterTask(void const * argument)
   while (1)
   {
     event = osMailGet(mail_box_id, 0);
-    if (event.status == osEventMail)
-    {
-      rptr = event.value.p;
+    rptr = event.value.p;
+    if (event.status == osEventMail && print_mode == PRINT_BLUETOOTH) {
```
Comment (johnyka) [inline Nucleo_F746ZG_app/Src/utility.c]: `event.status` is checked in every condition. You could make a separate `if` condition where you only check event.status and you make your print_mode checks inside

File: Nucleo_F746ZG_app/Src/main.c [hunk 3]
```diff
@@ -297,8 +299,11 @@ void HAL_UART_RxCpltCallback(UART_HandleTypeDef *huart)
     if(current_uart_char != '\0'){
       if(current_uart_char >= 'a' && current_uart_char <= 'z'){
         current_uart_char -= 32; //all to upper case
+      } else if(huart->Instance == USART6){
```
Comment (pVer297) [inline Nucleo_F746ZG_app/Src/main.c]: This should be "if(huart->Instance == USART3 || huart->Instance == USART6) {" you now checking if usart6 at the character modification

File: Nucleo_F746ZG_app/Src/utility.c [hunk 4]
```diff
@@ -27,12 +28,28 @@ void StartPrinterTask(void const * argument)
...
+        HAL_UART_Transmit(&huart3, (uint8_t*) rptr->array, strlen(rptr->array),
+            0xFFFF);
+      }
+      else if (print_mode == PRINT_BOTH)
+      {
+        HAL_UART_Transmit(&huart3, (uint8_t*) rptr->array, strlen(rptr->array),
+            0xFFFF);
+        HAL_UART_Transmit(&huart6, (uint8_t*) rptr->array, strlen(rptr->array),
+            0xFFFF);
+      }
```
Comment (pVer297) [inline Nucleo_F746ZG_app/Src/utility.c]: place the 0xFFFF back to the transmit line at every transmit

**Pull request closed without merging**